src/tbp/plot/plots/interactive_hypothesis_space_pointcloud.py

```diff
@@ -0,0 +1,1717 @@
+# Copyright 2025 Thousand Brains Project
+#
+# Copyright may exist in Contributors' modifications
+# and/or contributions to the work.
+#
+# Use of this source code is governed by the MIT
+# license that can be found in the LICENSE file or at
+# https://opensource.org/licenses/MIT.
+
+from __future__ import annotations
+
+import bisect
+import logging
+from copy import deepcopy
+from pathlib import Path
+from typing import TYPE_CHECKING, Any
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+from matplotlib import transforms
+from matplotlib.lines import Line2D
+from matplotlib.patches import Rectangle
+from pubsub.core import Publisher
+from vedo import Button, Image, Line, Mesh, Plotter, Points, Slider2D, Sphere
+
+from tbp.interactive.colors import Palette
+from tbp.interactive.data import (
+    DataLocator,
+    DataLocatorStep,
+    DataParser,
+    PretrainedModelsLoader,
+    YCBMeshLoader,
+)
+from tbp.interactive.topics import TopicMessage, TopicSpec
+from tbp.interactive.utils import (
+    Location3D,
+)
+from tbp.interactive.widget_updaters import WidgetUpdater
+from tbp.interactive.widgets import (
+    VtkDebounceScheduler,
+    Widget,
+    extract_button_state,
+    extract_slider_state,
+    set_button_state,
+    set_slider_state,
+)
+from tbp.plot.registry import attach_args, register
+
+if TYPE_CHECKING:
+    import argparse
+    from collections.abc import Callable, Iterable
+
+logger = logging.getLogger(__name__)
+
+FONT = "Arial"
+FONT_SIZE = 25
+
+
+class StepMapper:
+    """Bidirectional mapping between global step indices and (episode, local_step).
+
+    Global steps are defined as the concatenation of all local episode steps:
```

Comment:
Out of interest, did you manage to identify the cause of the issue where, after a jump-to-goal-state, the agent seemingly moves to a new location, while the sensory location remains un-updated?